VideoLocker/src/main/java/org/edx/mobile/view/adapters/InfiniteScrollUtils.java

```diff
@@ -154,11 +173,21 @@ private boolean isAbandoned() {
 
         @Override
         public void reset() {
-            activeLoadId.incrementAndGet(); // To disregard any in-progress loads
+            initLoading();
             adapter.clear();
+            onLoadMore();
+        }
+
+        @Override
+        public void refresh() {
```

Comment:
I think naming this `resetSilently()` would be more accurate, since it _is_ a destructive operation (discarding the pagination). But the final naming is up to you.